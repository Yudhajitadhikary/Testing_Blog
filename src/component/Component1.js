
import React,{useState} from 'react';

function Component1(props) {
    const [authors, setAuthors] = useState([]);
    //    const [activeAuthor, setActiveAuthor] = useState(null);
         const [posts, setPosts] = useState([]);
         //     // Load authors on start
    React.useEffect(() => {
      setAuthors([{id:1,name:'Yudhajit Adhikary',post:'Blog'},{id:1,name:'Koyena Adhikary',post:'Blog1'}])
      setPosts([{id:1,name:'Yudhajit Adhikary',post:'Blog'},{id:1,name:'Koyena Adhikary',post:'Blog1'}])
     },[]);
  
   
      
    return (
              <div style={{display:'flex'}}>
              <div>{authors&&<div>{authors.map((author,index)=>(<div>{author.name}</div>))}</div>}</div>
              <div>{posts&&<div>{posts.map((post,index)=>(<div>{post.post}</div>))}</div>}</div>
              </div>
              
            );
    }
    export default Component1;      

