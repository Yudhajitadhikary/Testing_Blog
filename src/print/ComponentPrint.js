import React, { Component } from 'react'
// import { connect } from 'react-redux';
export default class CardPrint extends Component {
    render() {
        const {email,password,pincode,address,phoneno}=this.props
        return (
            <div>
                <div style={{textAlign:'center',paddingLeft:'30px',fontSize:'30px',color:'green'}}>Here's a print copy of data you entered :)</div>
                <div style={{display:'flex',padding:'20px 20px 20px 50px '}} >
                    <div style={{color:'Red',paddingRight:'10px',fontSize:'22px'}}>Email:</div><div style={{fontSize:'22px'}}>{email}</div>
                </div>
                <div style={{display:'flex' ,padding:'20px 20px 20px 50px '}}>
                <div style={{color:'Red',paddingRight:'10px',fontSize:'22px'}}>Password:</div><div style={{fontSize:'22px'}}>{password}</div>
                </div >
                <div style={{display:'flex',padding:'20px 20px 20px 50px '}}>
                 <div style={{color:'Red',paddingRight:'10px',fontSize:'22px'}}>Pincode:</div><div style={{fontSize:'22px'}}>{pincode}</div>
                </div>
                <div style={{display:'flex' ,padding:'20px 20px 20px 50px '}}>
                 <div style={{color:'Red',paddingRight:'10px',fontSize:'22px'}}>Address:</div><div style={{fontSize:'22px'}}>{address}</div> 
                </div>
                <div style={{display:'flex' ,padding:'20px 20px 20px 50px '}}>
                   <div style={{color:'Red',paddingRight:'10px',fontSize:'22px'}}>PhoneNo:</div><div style={{fontSize:'22px'}}>{phoneno}</div> 
                </div>
            </div>
        )
    }
}

 
 
