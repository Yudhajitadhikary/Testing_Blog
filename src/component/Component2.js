import React, { Component } from 'react';
import Component1 from './Component1'
class Component2 extends Component {
    constructor(props) {
        super(props);
       
        this.Click = this.Click.bind(this);
        }
        Click() {
           this.props.action()
          }
    render() {
        const {count}=this.props
        return (
            <div>
            <button id='button' onClick={this.Click}>Count Total</button>
               <div>{count}</div>
               <Component1/>
            </div>
        );
    }
}

export default Component2;