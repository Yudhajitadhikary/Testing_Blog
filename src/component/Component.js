import React, { Component } from "react";
import ReactToPrint from "react-to-print";
import CardPrint from "../print/ComponentPrint";
import Component2 from "./Component2";
import { Formik, Field, Form, ErrorMessage } from "formik";
import * as Yup from "yup";
import { submitAction } from "../actions";
import { connect } from "react-redux";
class Card extends Component {
  constructor(props) {
    super(props);
    this.state = { value: "", count: 0, childCount: 0 };
    this.handleClick = this.handleClick.bind(this);
    this.childrenClick = this.childrenClick.bind(this);
  }
  handleClick() {
    this.setState({ count: this.state.count + 1 });
    document.getElementById("testId").style.display = "";
  }
  childrenClick() {
    this.setState({ childCount: this.state.childCount + 1 });
    document.getElementById("testId").style.display = "";
  }
  componentDidUpdate(PrevState) {
    console.log("Component Updated");
  }
  componentWillMount() {
    console.log("Component will mount");
  }
  componentDidMount() {
    console.log("Component did Mount");
  }

  render() {
    const {
      email,
      password,
      submitAction,
      phoneno,
      address,
      pincode,
    } = this.props;

    return (
      <div>
        <ReactToPrint
          trigger={() => (
            <button
              style={{
                textAlign: "center",
                marginLeft: "50%",
                backgroundColor: "lemonchiffon",
                width: "62px",
              }}
            >
              Print
            </button>
          )}
          content={() => this.componentRef}
        />
        <div style={{ display: "none" }}>
          <CardPrint
            email={email}
            password={password}
            phoneno={phoneno}
            address={address}
            pincode={pincode}
            ref={(el) => (this.componentRef = el)}
          />
        </div>
        <div>
          {" "}
          <div id="testId">{this.state.childCount}</div>
          <div id="testOnClick" onClick={this.handleClick}>
            Total Count
          </div>
        </div>
        <Component2 count={this.state.count} action={this.childrenClick} />
        <div>
          <Formik
            initialValues={{
              email: "",
              password: "",
              confirmPassword: "",
              pincode: "",
              address: "",
              phoneno: "",
            }}
            validationSchema={Yup.object().shape({
              email: Yup.string()
                .email("Email is invalid")
                .required("Email is required"),
              password: Yup.string()
                .min(6, "Password must be at least 6 characters")
                .matches(/[A-Z]{2}\d{2}[A-Z]{2}\d{4}$/i, "invalid Password")
                .required("Password is required"),
              confirmPassword: Yup.string()
                .oneOf([Yup.ref("password"), null], "Passwords must match")
                .required("Confirm Password is required"),
              pincode: Yup.string()
                // .max(6,'Pincode must be of 6 characters')
                .matches(/^[0-9]{6}$/, "Should be a number of length 6")
                .required("Pincode is required"),
              phoneno: Yup.string()
                // .max(6,'Pincode must be of 6 characters')
                .matches(/^[0-9]{10}$/, "Should be a number of length 10")
                .required("Contact Number is required"),
              address: Yup.string().required("Address is Required"),
            })}
            onSubmit={(fields) => {
              // console.log(fields)
              submitAction(fields);
            }}
            render={({ errors, status, touched }) => (
              <Form>
                <div className="form-group">
                  <label htmlFor="email">Email</label>
                  <Field
                    name="email"
                    type="text"
                    className={
                      "form-control" +
                      (errors.email && touched.email ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="email"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="password">Password</label>
                  <Field
                    name="password"
                    type="password"
                    className={
                      "form-control" +
                      (errors.password && touched.password ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="password"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="confirmPassword">Confirm Password</label>
                  <Field
                    name="confirmPassword"
                    type="password"
                    className={
                      "form-control" +
                      (errors.confirmPassword && touched.confirmPassword
                        ? " is-invalid"
                        : "")
                    }
                  />
                  <ErrorMessage
                    name="confirmPassword"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="address">Address</label>
                  <Field
                    name="address"
                    type="address"
                    className={
                      "form-control" +
                      (errors.address && touched.address ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="address"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="pincode">Pincode</label>
                  <Field
                    name="pincode"
                    type="pincode"
                    className={
                      "form-control" +
                      (errors.pincode && touched.pincode ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="pincode"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="phoneno">Contact Number</label>
                  <Field
                    name="phoneno"
                    type="phoneno"
                    className={
                      "form-control" +
                      (errors.phoneno && touched.phoneno ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="phoneno"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <button
                    type="submit"
                    style={{ marginLeft: "50%" }}
                    className="btn btn-primary mr-2"
                  >
                    Register
                  </button>
                </div>
              </Form>
            )}
          />
        </div>
      </div>
    );
  }
}
export const mapStateToProps = (state) => ({
  email: state.email,
  password: state.password,
  address: state.address,
  pincode: state.pincode,
  phoneno: state.phoneno,
});

export const mapDispatchToProps = (dispatch) => ({
  submitAction: (data) => dispatch(submitAction(data)),
});
export default connect(mapStateToProps, mapDispatchToProps)(Card);
