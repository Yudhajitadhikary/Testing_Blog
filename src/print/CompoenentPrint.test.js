import React from "react";
import { mount } from "../component/enzyme";
import { shallowToJson } from "enzyme-to-json";
import ComponentPrint from './ComponentPrint';
import { Provider } from 'react-redux';
import store from './../store';
 
let wrapper;
it("should render TextFieldAtom component", () => {
  wrapper = mount(<Provider store={store}><ComponentPrint/></Provider>);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
  

});

