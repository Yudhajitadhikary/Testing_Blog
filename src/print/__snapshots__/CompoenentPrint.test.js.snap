// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextFieldAtom component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <CardPrint>
    <div>
      <div
        style={
          Object {
            "color": "green",
            "fontSize": "30px",
            "paddingLeft": "30px",
            "textAlign": "center",
          }
        }
      >
        Here's a print copy of data you entered :)
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "padding": "20px 20px 20px 50px ",
          }
        }
      >
        <div
          style={
            Object {
              "color": "Red",
              "fontSize": "22px",
              "paddingRight": "10px",
            }
          }
        >
          Email:
        </div>
        <div
          style={
            Object {
              "fontSize": "22px",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "padding": "20px 20px 20px 50px ",
          }
        }
      >
        <div
          style={
            Object {
              "color": "Red",
              "fontSize": "22px",
              "paddingRight": "10px",
            }
          }
        >
          Password:
        </div>
        <div
          style={
            Object {
              "fontSize": "22px",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "padding": "20px 20px 20px 50px ",
          }
        }
      >
        <div
          style={
            Object {
              "color": "Red",
              "fontSize": "22px",
              "paddingRight": "10px",
            }
          }
        >
          Pincode:
        </div>
        <div
          style={
            Object {
              "fontSize": "22px",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "padding": "20px 20px 20px 50px ",
          }
        }
      >
        <div
          style={
            Object {
              "color": "Red",
              "fontSize": "22px",
              "paddingRight": "10px",
            }
          }
        >
          Address:
        </div>
        <div
          style={
            Object {
              "fontSize": "22px",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "padding": "20px 20px 20px 50px ",
          }
        }
      >
        <div
          style={
            Object {
              "color": "Red",
              "fontSize": "22px",
              "paddingRight": "10px",
            }
          }
        >
          PhoneNo:
        </div>
        <div
          style={
            Object {
              "fontSize": "22px",
            }
          }
        />
      </div>
    </div>
  </CardPrint>
</Provider>
`;
